marshal_10_decimal_test.go

```diff
@@ -0,0 +1,54 @@
+package gocql
+
+import (
+	"gopkg.in/inf.v0"
+	"math"
+	"testing"
+
+	"github.com/gocql/gocql/internal/tests/utils"
+	"github.com/gocql/gocql/marshal/tests/mod"
+	"github.com/gocql/gocql/marshal/tests/serialization"
+)
+
+func TestMarshalDecimal(t *testing.T) {
+	marshal := func(i interface{}) ([]byte, error) { return Marshal(NativeType{proto: 4, typ: TypeDecimal}, i) }
+	unmarshal := func(bytes []byte, i interface{}) error {
+		return Unmarshal(NativeType{proto: 4, typ: TypeDecimal}, bytes, i)
+	}
+
+	// In cases with data len <4 unmarshal returns an error: "inf.Dec needs at least 4 bytes, while value has only 0"
+	// So, unmarshalls 'nil' and 'zero' 'decimal' currently impossible.
```

Comment:
```suggestion
	// Unmarshal does not support deserialization of 'nil' and 'zero' 'into `inf.Dec`.
```

Reply:
@illia-li , i don't see technical problem to make it happen, can I ask you either create an issue to fix that or outline why it is better not to do it

Reply:
Done in this version:
// Unmarshal does not support deserialization of `decimal` with `nil` and `zero` `value len` 'into `inf.Dec`.

Reply:
What about issue ? are you going to create it ?

Reply:
in progress

Reply:
Issue done #282